posts/2024-08-26-webassembly-targets-change-in-default-target-features.md

```diff
@@ -128,6 +164,30 @@ Supporting WebAssembly multi-return functions in Rust is a broader topic than
 this post can cover, but at this time it's an area that's ripe for contribution
 from suitably motivated contributors.
 
+### Aside: ABI Stability and WebAssembly
+
+While on the topics of ABIs and the `multi-value` feature it's perhaps worth
+also going over a bit what ABIs mean for WebAssembly. Currently there's
+effectively only one ABI implemented by LLVM and the Rust compiler meaning that
+at the WebAssembly module level `extern "C"` and `extern "Rust"` are
```

Comment:
`extern "Rust"` doesn't have the struct unpacking behavior that `extern "C"` has on wasm32-unknown-unknown. And `extern "Rust"` passes small structs as up to two arguments, while `extern "C"` on wasm32-wasi and wasm32-unknown-emscripten would pass structs by-ref if they have more than a single primitive field.

Reply:
That's an excellent point and reminds me to be more precise here. I tend to always forget "ABI" additionally refers to passing structures by value which means my own communication about this isn't always precise as how you pass structures by value is rightfully a huge part of how ABIs work. I've reworded this section to remove saying that `extern "C"` is the same as `extern "Rust"` but still explain how `extern "C"` should match Clang and is hard to change while changing `extern "Rust"` should be possible with LLVM support.